update/delta/delta_1.3.7_000_search_path.sql

```diff
@@ -0,0 +1,456 @@
+        from
+			(select 
+				g.id,
+				-- si le tronçon suivant a la même target que le tronçon que l'on vient de parcourir, 
+				-- il faut inverser et prendre cette target comme source, sinon on prend la source
+				case 
+					when sg.target=g.target then g.target
+					else g.source
+				end as source,
+				-- si le tronçon suivant a la même target que le tronçon que l'on vient de parcourir, 
+				-- il faut inverser et prendre la source comme target, sinon on prend la target
+				case 
+					when sg.target=g.target then g.source
+					else g.target
+				end as target,
+				g.cost,
+				sg.depth,
+				sg.path
+			from 
+            -- la table qu'on jointure : c'est le graphe (au format pgrouting)
+				qwat_network.network as g,
+			--join
+            -- la CTE
+				search_graph as sg				
+			where
+			--on
+				-- on jointure la CTE et network pour se déplacer dans le graphe
+				-- on cherche le tronçon qui a la target du tronçon précédent comme source
+				-- (ou comme target si le réseau n'a pas été saisi correctement et on inverse ensuite)		
+				(sg.target = g.source or
+				sg.target = g.target
+				and sg.source <> g.source)
+			) as ng
+
+        where
+        	-- on continue tant qu'on n'est pas au noeud d'arrivée
+			(ng.source <> end_node)
+			-- on ne repasse pas par un noeud déjà dans le chemin à parcourir (on ne revient pas en arrière)
+			and not ng.target = any(ng.path)
+			-- on s'arrête à une profondeur pour ne pas parcourir le réseau entier
+			and ng.depth < max_depth
+            -- et on ne passe pas par les vannes fermées
+            --and sg.cost != -1            
+	)
+	-- on met le chemin dans une variable pour construire ensuite le linéaire avec les géométries
+    select sg.path into path
+		from search_graph as sg
+ 		where sg.target = end_node
+		order by sg.depth, sg.source, sg.target
+		limit 1;
+    
+	if path is not null then
+		-- on prend le premier noeud du chemin (noeud de départ)
+		node_0 := path[1];
+		-- on part ensuite du deuxième noeud pour construire chaque géométrie
+	    foreach node in array path[2:]
+		loop
+			select n.id, n.source, n.target, n.geometry into id, source, target, geometry from qwat_network.network n
+			-- on cherche le tronçon avec source et target égales au noeud n et noeud n + 1
+			-- (ou l'inverse si les tronçons n'ont pas été saisis correctement)
+	    	where (n.source = node_0 and n.target = node) or (n.target = node_0 and n.source = node);
+	--   	    raise notice 'node 0 % ; node %', node_0, node;
+			-- on passe au noeud suivant pour chercher le prochain tronçon
+	    	node_0 := node;
+	    	-- on renvoie la ligne de résultat dans la table
+	        return next;
+	--        raise notice 'node %', node;
+		end loop;
+	end if;	
+end
+$function$
+;
+
+select qwat_network.ft_all_pipes();
+
+select qwat_network.ft_create_network();
+
+CREATE OR REPLACE FUNCTION qwat_network.ft_network_cutoff(valves integer[], pipe integer,  max_km integer DEFAULT 20)
```

Comment:
there seems to be a code redundancy with `qwat_network.ft_search_path`, could this be grouped somehow?